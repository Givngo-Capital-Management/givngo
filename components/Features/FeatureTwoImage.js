import React from 'react';
import styles from './FeatureTwoImage.module.scss';

export default function FeatureTwoImage({ urls, position, srcsetParam }) {
  return (
    <div className={styles.section}>
      <div className={styles['div-part-1']}>
        <div className={styles['div-wrap-1']}>
          <div
            className={styles['text-box']}
            style={
              position === 'left'
                ? { paddingLeft: '100px', marginLeft: 'auto' }
                : { paddingRight: '100px' }
            }
          >
            <h1 className={styles['heading-1']}>The donation sensation</h1>
            <p className={styles['paragraph-1']}>
              Ginvgoâ€™s mission is to help nonprofits and charities raise
              donations. We have built a platform so people can quickly learn
              about a cause or charity and then donate and support. We have
              eliminated as many steps as possible in the donation process which
              generates the highest conversions possible.
            </p>
          </div>
          <div className={styles['image-wrap']}>
            <img
              src={urls[0]}
              width="1280"
              srcSet={srcsetParam}
              sizes="(max-width: 479px) 92vw, (max-width: 767px) 95vw, 96vw"
              alt=""
              className={styles['image-1']}
            />
          </div>
        </div>
      </div>
      <div
        className={styles['div-part-2']}
        style={
          position === 'left'
            ? { justifyContent: 'flex-start' }
            : { justifyContent: 'flex-end' }
        }
      >
        <div
          className={styles.image}
          style={{ backgroundImage: `url(${urls[1]})` }}
        />
      </div>
    </div>
  );
}
