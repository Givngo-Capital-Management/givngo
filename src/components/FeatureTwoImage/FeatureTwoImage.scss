.section {
  position: relative;
  padding-top: 100px;
  padding-bottom: 100px;

  .div-part-1 {
    max-width: 1280px;
    margin-right: auto;
    margin-left: auto;
    padding-top: 0px;
    padding-bottom: 0px;
    border-bottom-style: none;
    border-radius: 1px;

    .div-wrap-1 {
      padding-right: 20px;
      padding-left: 20px;
      -webkit-box-orient: vertical;
      -webkit-box-direction: normal;
      flex-direction: column;
    
      .text-box {
        position: relative;
        z-index: 1;
        max-width: 50%;
        margin-bottom: 100px;
        padding-right: 100px;
        -webkit-box-flex: 1;
        flex: 1;
      
        
        .heading-1 {
          margin-top: 0px;
          margin-bottom: 40px;
          font-size: 40px;
          line-height: 48px;
        }

        .paragraph-1 {
          font-size: 20px;
          line-height: 30px;
        }    
      }
      
      .image-wrap {
        position: relative;
        z-index: 1;
        display: flex;
        -webkit-box-pack: start;
        justify-content: flex-start;
        -webkit-box-flex: 1;
        flex: 1;

        .image-1 {
          overflow: hidden;
          border-radius: 4px;
          box-shadow: 0 0 25px 0 rgba(0, 0, 0, 0.2);
        }
      }
    }
  }

  .div-part-2 {
    position: absolute;
    left: 0%;
    top: 0%;
    right: 0%;
    bottom: 0%;
    z-index: 0;
    display: flex;
    overflow: hidden;
    padding-top: 100px;
    -webkit-box-pack: end;
    justify-content: flex-end;

    .image {
      position: relative;
      overflow: hidden;
      width: 50%;
      height: 600px;
      border-radius: 4px;
      background-image: url(https://givngo.io/wp-content/themes/givngo/images/2305-6-Eblast-One-Tree-Planted-One-Tree-Planted-2.jpg);
      background-position: 35% 50%;
      background-size: cover;
    }
  }
}
