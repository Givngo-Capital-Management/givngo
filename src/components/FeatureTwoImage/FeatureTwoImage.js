import React from 'react'
import './FeatureTwoImage.scss'

export default function FeatureTwoImage({urls, position, srcset_param}) {
  return (
    <div className="section">
      <div className="div-part-1">
        <div className="div-wrap-1">
          <div className="text-box" style={position==="left" ? {paddingLeft: "100px", marginLeft: "auto"}:{paddingRight: "100px"}}>
            <h1 className="heading-1">The donation sensation</h1>
            <p className="paragraph-1">Ginvgoâ€™s mission is to help nonprofits and charities raise donations. We have built a platform so people can quickly learn about a cause or charity and then donate and support. We have eliminated as many steps as possible in the donation process which generates the highest conversions possible.</p>
          </div>
          <div className="image-wrap">
            <img 
              src={urls[0]} 
              width="1280" 
              srcset={srcset_param}
              sizes="(max-width: 479px) 92vw, (max-width: 767px) 95vw, 96vw" 
              alt="" 
              className="image-1">
            </img>
          </div>
        </div>
      </div>
      <div className="div-part-2" style={position==="left" ? {justifyContent: "flex-start"}:{justifyContent: "flex-end"}}>
        <div className="image" style={{backgroundImage: "url(" + urls[1] + ")"}}></div>
      </div>
    </div>
  );
}