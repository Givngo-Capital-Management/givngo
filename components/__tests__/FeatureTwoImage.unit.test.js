import React from 'react';
import { shallow } from 'enzyme';
import FeatureTwoImage from '../Features/FeatureTwoImage';

const images = [
  'https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII.png',
  'https://givngo.io/wp-content/themes/givngo/images/2305-6-Eblast-One-Tree-Planted-One-Tree-Planted-2.jpg',
];

const srcsetParam =
  'https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-500.png 500w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-800.png 800w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-1080.png 1080w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII.png 2560w';

describe('FeatureTwoImage', () => {
  test('render without props', () => {
    const wrapper = shallow(<FeatureTwoImage />);
    expect(wrapper).toMatchSnapshot();
  });
  test('renders on left', () => {
    const wrapper = shallow(
      <FeatureTwoImage
        urls={images}
        position="left"
        srcsetParam={srcsetParam}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
  test('renders on right', () => {
    const wrapper = shallow(
      <FeatureTwoImage
        urls={images}
        position="right"
        srcsetParam={srcsetParam}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
