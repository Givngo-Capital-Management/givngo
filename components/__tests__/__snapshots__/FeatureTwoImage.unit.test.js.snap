// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeatureTwoImage renders on left 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "marginLeft": "auto",
            "paddingLeft": "100px",
          }
        }
      >
        <h1>
          The donation sensation
        </h1>
        <p>
          Ginvgo’s mission is to help nonprofits and charities raise donations. We have built a platform so people can quickly learn about a cause or charity and then donate and support. We have eliminated as many steps as possible in the donation process which generates the highest conversions possible.
        </p>
      </div>
      <div>
        <img
          alt=""
          sizes="(max-width: 479px) 92vw, (max-width: 767px) 95vw, 96vw"
          src="https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII.png"
          srcSet="https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-500.png 500w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-800.png 800w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-1080.png 1080w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII.png 2560w"
          width="1280"
        />
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "justifyContent": "flex-start",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://givngo.io/wp-content/themes/givngo/images/2305-6-Eblast-One-Tree-Planted-One-Tree-Planted-2.jpg)",
        }
      }
    />
  </div>
</div>
`;

exports[`FeatureTwoImage renders on right 1`] = `
<div>
  <div>
    <div>
      <div
        style={
          Object {
            "paddingRight": "100px",
          }
        }
      >
        <h1>
          The donation sensation
        </h1>
        <p>
          Ginvgo’s mission is to help nonprofits and charities raise donations. We have built a platform so people can quickly learn about a cause or charity and then donate and support. We have eliminated as many steps as possible in the donation process which generates the highest conversions possible.
        </p>
      </div>
      <div>
        <img
          alt=""
          sizes="(max-width: 479px) 92vw, (max-width: 767px) 95vw, 96vw"
          src="https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII.png"
          srcSet="https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-500.png 500w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-800.png 800w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII-p-1080.png 1080w, https://givngo.io/wp-content/themes/givngo/images/OOOTTTIIIIII.png 2560w"
          width="1280"
        />
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "justifyContent": "flex-end",
      }
    }
  >
    <div
      style={
        Object {
          "backgroundImage": "url(https://givngo.io/wp-content/themes/givngo/images/2305-6-Eblast-One-Tree-Planted-One-Tree-Planted-2.jpg)",
        }
      }
    />
  </div>
</div>
`;
